airflow/utils/log/file_task_handler.py

```diff
@@ -168,6 +168,11 @@ def _change_directory_permissions_up(directory: Path, folder_permissions: int):
     :param directory: directory to change permissions of (including parents)
     :param folder_permissions: permissions to set
     """
+    if directory.resolve() == Path.home().resolve():
```

Comment:
I think about other case, maybe not really common but I guess pretty popular, run under `root` ⚠️⚠️⚠️  user

In linux usually home directory is `/root` and lets imagine that logs stored into the `/opt/airflow/logs/foo/bar`, so lookup would be

- `/opt/airflow/logs/foo/bar`
- `/opt/airflow/logs/foo`
- `/opt/airflow/logs`
- `/opt/airflow`
- `/opt`
- `/`

Reply:
Yeah. Crossed my mind. But then - we have no way to prevent it.

Reply:
Maybe check if the user is a root `os.geteuid() == 0` and do nothing, except maybe huge banner that do not run under the root.

Reply:
I found a better way. I only change permissions for the directories that have **just** been created (I stop at the directory that was existing before mkdir). This is way better and correct.